pkg/configuration/filter_resources.go

```diff
@@ -0,0 +1,141 @@
+	vnisAttIds          []string // the vni attach id fo the filtered vnis, use to filter the segment ports
+	vmsAddresses        []string // the vm addresses of the filtered the vms
+	allGroupPaths       []string // all the group paths
+	allRemainGroupPaths []string // the filtered paths
+}
+
+func (f *resourceFilter) filterTopology() {
+	// removing vms from vm list:
+	f.rc.VirtualMachineList = slices.DeleteFunc(f.rc.VirtualMachineList, f.vmFilter)
+	f.vmIds = common.CustomStrSliceToStrings(f.rc.VirtualMachineList, func(vm collector.VirtualMachine) string { return *vm.ExternalId })
+	// remove vnis from list:
+	f.rc.VirtualNetworkInterfaceList = slices.DeleteFunc(f.rc.VirtualNetworkInterfaceList, f.vniFilter)
+	f.vnisAttIds = common.CustomStrSliceToStrings(f.rc.VirtualNetworkInterfaceList,
+		func(vni collector.VirtualNetworkInterface) string { return *vni.LportAttachmentId })
+	// remove segment ports:
+	for i := range f.rc.SegmentList {
+		segment := &f.rc.SegmentList[i]
+		segment.SegmentPorts = slices.DeleteFunc(segment.SegmentPorts, f.portFilter)
+	}
+	// remove empty segments:
+	f.rc.SegmentList = slices.DeleteFunc(f.rc.SegmentList, f.segmentFilter)
+}
+
+func (f *resourceFilter) filterGroups() {
+	// calc addresses of all vms:
+	f.vmsAddresses = common.CustomStrsSliceToStrings(f.rc.VirtualNetworkInterfaceList, func(vni collector.VirtualNetworkInterface) []string {
+		return common.CustomStrsSliceToStrings(vni.IpAddressInfo, func(info nsx.IpAddressInfo) []string {
+			return common.CustomStrSliceToStrings(info.IpAddresses, func(ip nsx.IPAddress) string { return string(ip) })
+		})
+	})
+	// remove filtered vms, vnis and addresses from groups:
+	for i := range f.rc.DomainList {
+		domainRsc := &f.rc.DomainList[i].Resources
+		f.allGroupPaths = append(f.allGroupPaths, common.CustomStrSliceToStrings(domainRsc.GroupList,
+			func(group collector.Group) string { return *group.Path })...)
+		for j := range domainRsc.GroupList {
+			domainRsc.GroupList[j].VMMembers = slices.DeleteFunc(domainRsc.GroupList[j].VMMembers, f.groupVMFilter)
+			domainRsc.GroupList[j].VIFMembers = slices.DeleteFunc(domainRsc.GroupList[j].VIFMembers, f.vniFilter)
+			domainRsc.GroupList[j].AddressMembers = slices.DeleteFunc(domainRsc.GroupList[j].AddressMembers, f.addressFilter)
+		}
+		// remove empty groups:
+		domainRsc.GroupList = slices.DeleteFunc(domainRsc.GroupList, f.groupFilter)
+		f.allRemainGroupPaths = append(f.allRemainGroupPaths, common.CustomStrSliceToStrings(domainRsc.GroupList,
+			func(group collector.Group) string { return *group.Path })...)
+	}
+}
+func (f *resourceFilter) filterRules() {
+	// handling rules:
+	for i := range f.rc.DomainList {
+		domainRsc := f.rc.DomainList[i].Resources
+		for j := range domainRsc.SecurityPolicyList {
+			secPolicy := &domainRsc.SecurityPolicyList[j]
+			secPolicy.Scope = slices.DeleteFunc(secPolicy.Scope, f.groupPathFilter)
+			for i := range secPolicy.Rules {
+				rule := &secPolicy.Rules[i]
```

Comment:
I would print to debug log the original rule and the new rule for every rule changed here..